lib/trie/trie.go

```diff
@@ -521,6 +518,7 @@ func (t *Trie) insertInBranch(parentBranch *Node, key, value []byte,
 			return nil, false, 0, fmt.Errorf("preparing branch for mutation: %w", err)
 		}
 		parentBranch.StorageValue = value
+		parentBranch.HashedValue = isValueHashed
```

Comment:
nit: maybe make it `node.IsHashedValue` to know that it's a boolean. 
```suggestion
		parentBranch.IsHashedValue = isValueHashed
```